### PR TITLE
Remove ObservableBoard and BoardObserver

### PR DESCRIPTION
**Pull request by hongshui**
There was issue with StackOverflow error and removing those two interfaces and replacing them with while loop resolved the issue.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -14,17 +14,19 @@
         <project.java.version>11</project.java.version>
         <maven.compiler.release>${project.java.version}</maven.compiler.release>
 
+        <version.testing>7.1.0</version.testing>
+        <version.mockito>3.2.4</version.mockito>
```
Comment (KrzysiekJodlowski) [inline pom.xml]: Eventually "version.dependency.testing" (same with mockito) but its up to you. Also, "testing"? Not "testng"?

File: pom.xml [hunk 2]
```diff
@@ -43,6 +45,36 @@
...
+            </plugin>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-pmd-plugin</artifactId>
+                <version>3.12.0</version>
+            </plugin>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-checkstyle-plugin</artifactId>
+                <version>3.1.0</version>
```
Comment (KrzysiekJodlowski) [inline pom.xml]: Add versions to properties, stay consistent:)

File: src/main/java/com/github/hongshui/ox/board/Board.java [hunk 3]
```diff
@@ -12,16 +12,16 @@
 		this.borderLength = borderLength;
 	}
 
-	void insertSymbol(char symbol, Coordinates coordinates, boolean disableNotifications) {
+	public void insertSymbol(char symbol, Coordinates coordinates) {
```
Comment (KrzysiekJodlowski) [inline src/main/java/com/github/hongshui/ox/board/Board.java]: It's good that you've removed it (that boolean passing), I forgot to mention this one in last PR.

File: src/main/java/com/github/hongshui/ox/board/BoardMaker.java [hunk 4]
```diff
@@ -1,7 +1,5 @@
 package com.github.hongshui.ox.board;
 
-import com.github.hongshui.ox.commons.Coordinates;
-
 /**
  * This class is a gate to board package.
  * It allows to generate new board (with observer) and to update it.
```
Comment (KrzysiekJodlowski) [inline src/main/java/com/github/hongshui/ox/board/BoardMaker.java]: "with observer" - is it relevant now?

File: src/main/java/com/github/hongshui/ox/datasource/InteractiveDataSource.java [hunk 5]
```diff
@@ -53,7 +53,8 @@ private void invalidMode() {
 	@Override
 	public int acquireBorderLength() {
 		ui.print(SELECT_BORDER);
-		return parseInt(validatedInput(validator::isValidBorder, this::invalidBorderMessage));
+		return parseInt(
+				validatedInput(validator::isValidBorder, this::invalidBorderMessage));
```
Comment (KrzysiekJodlowski) [inline src/main/java/com/github/hongshui/ox/datasource/InteractiveDataSource.java]: Is it really shorter than before?;)

File: src/main/java/com/github/hongshui/ox/datasource/InteractiveDataSource.java [hunk 6]
```diff
@@ -63,7 +64,8 @@ private void invalidBorderMessage() {
 	@Override
 	public int acquireWinCondition() {
 		ui.print(SELECT_WIN_CONDITION);
-		return parseInt(validatedInput(validator::isValidWinCondition, this::invalidWinConditionMessage));
+		return parseInt(
+				validatedInput(validator::isValidWinCondition, this::invalidWinConditionMessage));
```
Comment (KrzysiekJodlowski) [inline src/main/java/com/github/hongshui/ox/datasource/InteractiveDataSource.java]: Same as above.

File: src/main/java/com/github/hongshui/ox/datasource/RandomDataSource.java [hunk 7]
```diff
@@ -49,6 +49,6 @@ private int visiblySelect(int number) {
 
 	void visiblyWait() {
 		try { TimeUnit.MILLISECONDS.sleep(400); }
-		catch(Exception e) { System.exit(0); }
+		catch(Exception e) { ui.print(""); }
```
Comment (KrzysiekJodlowski) [inline src/main/java/com/github/hongshui/ox/datasource/RandomDataSource.java]: It's good that you've removed this System.exit!

File: src/main/java/com/github/hongshui/ox/game/Game.java [hunk 8]
```diff
@@ -42,7 +42,8 @@ private void playTheGame() {
 		IntStream
 				.range(0, NUMBER_OF_ROUNDS)
 				.mapToObj(i -> new Match(
-						ui, players, setup.newDataSource(border), boardMaker, new BoardPainter(border)))
+						ui, players, setup.newDataSource(border),
+						boardMaker.generateNewBoard(), new BoardPainter(border)))
```
Comment (KrzysiekJodlowski) [inline src/main/java/com/github/hongshui/ox/game/Game.java]: Empty line.

File: src/main/java/com/github/hongshui/ox/game/Match.java [hunk 9]
```diff
@@ -10,65 +9,66 @@
...
+			Coordinates coordinates = nextMove();
+			isWin = board.isWin(coordinates);
+			isDraw = board.isDraw();
+			paintBoard();
+			if (!isDraw && !isWin) { players.nextPlayer(); }
+		}
+		while (!isDraw && !isWin);
+
+		if (isWin) { onWin(); }
+		else { onDraw(); }
```
Comment (KrzysiekJodlowski) [inline src/main/java/com/github/hongshui/ox/game/Match.java]: That looks strange;) But its up to you:)

File: pom.xml (hunk 1 above)
Comment (hongshui) [reply pom.xml]: Done :)

File: pom.xml (hunk 2 above)
Comment (hongshui) [reply pom.xml]: Done :)

File: src/main/java/com/github/hongshui/ox/board/Board.java (hunk 3 above)
Comment (hongshui) [reply src/main/java/com/github/hongshui/ox/board/Board.java]: :)

File: src/main/java/com/github/hongshui/ox/board/BoardMaker.java (hunk 4 above)
Comment (hongshui) [reply src/main/java/com/github/hongshui/ox/board/BoardMaker.java]: No, I will delete it

File: src/main/java/com/github/hongshui/ox/datasource/InteractiveDataSource.java (hunk 5 above)
Comment (hongshui) [reply src/main/java/com/github/hongshui/ox/datasource/InteractiveDataSource.java]: In Intelij yes

File: src/main/java/com/github/hongshui/ox/datasource/RandomDataSource.java (hunk 7 above)
Comment (hongshui) [reply src/main/java/com/github/hongshui/ox/datasource/RandomDataSource.java]: Though current solution is still a bit of cheating but who cares ;)

**Pull request merged**